compiler/src/main/kotlin/se/ansman/kotshi/Property.kt

```diff
@@ -53,7 +56,8 @@ class Property(
     init {
         require(getter != null || field != null)
 
-        defaultValueProvider = if (defaultValueQualifier != null || parameter.hasAnnotation<JsonDefaultValue>()) {
+        defaultValueProvider = if (isTransient || defaultValueQualifier != null ||
```

Comment:
We should make sure that the error message is understandable in this case since it's implied that you are using default values.